Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -291,3 +385,224 @@ private extension CardReaderConfigError {
         }
     }
 }
+
+// MARK: Networking Methods
+private extension CardPresentPaymentStore {
+    /// Switch the store to use WCPay as the backend.
+    /// Does nothing if the store is already using WCPay.
+    /// WCPay is the initial default for the store.
+    ///
+    func useWCPayAsBackend() {
+        guard usingBackend != .wcpay else {
+            return
+        }
+
+        reset() // Concern: asynchronicity
```

Comment:
I am not sure I would call reset here. If we do, the API docs of these methods should, in my opinion, clarify that there is going to be an implicit call to reset.

I think it would be slightly preferable to leave the responsibility of resetting the SDK to an upper layer.

Also, the Terminal SDK tends to error quite often for attempting operations when it's busy, so we should make sure we do not introduce instability

Reply:
I've removed the resets from the service layer in https://github.com/woocommerce/woocommerce-ios/pull/5868/commits/fc0132753e9c15b45801a177b871b6d904fdc9a1 but I haven't added them to the upper layer yet. During testing I noticed that even when switching between WCPay powered and Stripe powered stores, the Terminal SDK was continuing to ask the previous service for tokens and location.

I'll continue digging into this. I think we might have missed something in https://github.com/woocommerce/woocommerce-ios/pull/4147

Reply:
Could it because there is no call to reset() before switching the store?

Looking at the implementation of setConfigProvider in StripeCardReaderService, there is this:

```
        if !Terminal.hasTokenProvider() {
            Terminal.setTokenProvider(tokenProvider)
        }
```